WebDriverAgentLib/WebDriverCommands/FBElementCommands.m

```diff
@@ -191,9 +192,63 @@ + (NSArray *)routes
 
 + (void)typeText:(NSString *)text
 {
-  UIAKeyboard *keyboard = [[[UIATarget localTarget] frontMostApp] keyboard];
-  [keyboard setInterKeyDelay:0.25];
-  [keyboard typeString:text];
+  for (NSInteger i = 0; i < [text length]; i++) {
+    NSString *characterString = [text substringWithRange:NSMakeRange(i, 1)];
+    NSDictionary *shift =
+    @{
+      @"!": @"1",
+      @"@": @"2",
+      @"#": @"3",
+      @"$": @"4",
+      @"%": @"5",
+      @"^": @"6",
+      @"&": @"7",
+      @"*": @"8",
+      @"(": @"9",
+      @")": @"0",
+      @"_": @"-",
+      @"?": @"/",
+      @"+": @"=",
+      @":": @";",
+      @"~": @"`",
+      @"{": @"[",
+      @"}": @"]",
+      @"\"": @"'",
+      @"<": @",",
+      @">": @".",
+      @"~": @"`",
+      @"|": @"\\",
+      };
+    NSArray *specialKeyCodes = @[ @"-", @"=", @"[", @"]", @"\\", @"\\", @";", @"'", @"`", @",", @".", @"/" ];
```

Comment:
Special keys mapping changes depending on keyboard you are using, unfortunately. However `[keyboard typeString:text]` wouldn't work either so :)